packages/firestore/test/integration/api/validation.test.ts

```diff
@@ -1171,6 +1204,36 @@ apiDescribe('Validation:', (persistence: boolean) => {
         "Invalid query. You cannot use 'array-contains' filters with " +
           "'array-contains-any' filters."
       );
+
+      expect(() =>
+        db
+          .collection('test')
+          .where('foo', notInOp, [2, 3])
+          .where('foo', 'array-contains', 1)
+      ).to.throw(
+        "Invalid query. You cannot use 'array-contains' filters with " +
+          "'not-in' filters."
+      );
+    });
+
+    validationIt(persistence, 'with != and NOT_IN filters fail', db => {
```

Comment:
Since the user-visible spelling for NOT_IN is not-in, maybe describe this in terms of "not-in" rather than the proto name for this operator?